### PR TITLE
Fixed GetRoomById Endpoint to return proper room (not first found).

### PR DESCRIPTION
**Pull request by maydeff**

### CODE REVIEW

File: server/SmartHome.BusinessLogic/Components/CommandHandlers/SwitchDeviceCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+namespace SmartHome.BusinessLogic.Components.CommandHandlers
+{
+    public class SwitchDeviceCommandHandler
+    {
+        private readonly SmartHomeContext _context;
+        private readonly SmartHomeEntityExistsValidationRule _smartHomeEntityExistsValidationRule;
+        private readonly UserExistsValidationRule _userExistsValidationRule;
+        private readonly UserIsConnectedToSmartHomeEntityValidationRule _userIsConnectedToSmartHomeEntityValidationRule;
+        private readonly DeviceExistsValidationRule _deviceExistsValidationRule;
```
Comment (Snooking) [inline server/SmartHome.BusinessLogic/Components/CommandHandlers/SwitchDeviceCommandHandler.cs:19]: Missing device is in room validation rule

File: server/SmartHome.API/Rooms/RoomsController.cs [hunk 2]
```diff
@@ -54,7 +58,27 @@ public async Task<IActionResult> GetRoomById([FromQuery] Guid userId, [FromQuery
                 return result.ResultError.ToProperErrorResult();
             }
 
-            return new OkObjectResult(result);
+            return new OkObjectResult(result.Data);
         }
+
+        [HttpPut(Route + "/{roomId}/components/{componentId}")]
+        public async Task<IActionResult> SwitchDevice([FromQuery] UserIdSmartHomeEntityIdQueryParam queryParam, [FromRoute] Guid roomId, [FromRoute] Guid componentId, [FromQuery] bool isOn)
```
Comment (Snooking) [inline server/SmartHome.API/Rooms/RoomsController.cs]: isOn is not a proper name. I have no idea whether I'm supposed to give You current state or the desired state

File: server/SmartHome.API/Infrastructure/UserIdSmartHomeEntityIdQueryParam.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace SmartHome.API.Infrastructure
+{
+    public class UserIdSmartHomeEntityIdQueryParam
+    {
+        public Guid UserId { get; set; }
```
Comment (Snooking) [inline server/SmartHome.API/Infrastructure/UserIdSmartHomeEntityIdQueryParam.cs]: Please change the name to `RequestedByUserId`. Later, when we'll add changing other users permissions, we will have name conflict, so let's come prepared

**Review by Snooking — CHANGES_REQUESTED** [review]

File: server/SmartHome.API/Rooms/RoomsController.cs (hunk 2 above)
Comment (maydeff) [reply server/SmartHome.API/Rooms/RoomsController.cs]: I just follow endpoint document XD

File: server/SmartHome.BusinessLogic/Components/CommandHandlers/SwitchDeviceCommandHandler.cs (hunk 1 above)
Comment (maydeff) [reply server/SmartHome.BusinessLogic/Components/CommandHandlers/SwitchDeviceCommandHandler.cs:19]: done

File: server/SmartHome.API/Rooms/RoomsController.cs (hunk 2 above)
Comment (maydeff) [reply server/SmartHome.API/Rooms/RoomsController.cs]: changed to newState, i believe you will like it ;) also modified endpoint document

File: server/SmartHome.API/Infrastructure/UserIdSmartHomeEntityIdQueryParam.cs (hunk 3 above)
Comment (maydeff) [reply server/SmartHome.API/Infrastructure/UserIdSmartHomeEntityIdQueryParam.cs]: yep, done

**Review by Snooking — APPROVED** [review]

**Pull request merged**